### DISCUSSION TITLE
XMPP (RFC 6120), PubSub (XEP-0060) and Atom Over XMPP (RFC 4287)

### DISCUSSION THREAD

**Discussion by sjehuda** [FreshRSS/FreshRSS#6379] [Ideas]
Greetings!

My name is Schimon and I am a champion of Syndication and XMPP.

For those of you who are not famiiliar, XMPP (previously Jabber) is a messaging system with open standards. XMPP is extensible via XEP.

I will discuss about XEP-0060: Publish-Subscribe.

Almost a couple of decades ago, a [discussion](https://datatracker.ietf.org/meeting/66/materials/slides-66-atompub-1.pdf) arose about incorporating Atom (RFC 4287) with XMPP PubSub, which is really easy because both are XML data.

One of the major benefits of utilizing PubSub is native Push Notifications (yes, that technique was much before HTML5) which do not require HTTP pulling.

That would be a great feature to have FreshRSS to utilize XMPP PubSub natively!

See a [proof of concept](https://codeberg.org/poezio/slixmpp/issues/3526) I have made to parse PubSub nodes as Atom Syndication Format.

https://datatracker.ietf.org/meeting/66/materials/slides-66-atompub-1.pdf
https://codeberg.org/poezio/slixmpp/issues/3526

Cheers,
Schimon

**Comment by Alkarex** [comment]
Hello,
I will come back to read in more details later, but a quick answer already: we support WebSub https://www.w3.org/TR/websub/ (formerly PubSubHubbub). You can test it in FreshRSS by subscribing to e.g. http://push-tester.cweiske.de to demonstrate instant notifications without HTTP pulling.
Does that help?

**Reply by sjehuda** [reply]
Alkarex, thank you for your prompt respond!

There are two aspects/features to this discussion:

### 1) Downloading data from PubSub nodes.

This would allow feeds (i.e. nodes) that are updated upon action of nodes.

### 2) Storing data to PubSub nodes.

Storing data on XMPP would be beneficial for people who use XMPP  without a feed reader such as FreshRSS.

People who use XMPP will be able to view news aggregated by FreshRSS to be viewed from their own clients like Movim and Libervia.

Use case: I want to share news with friends of my groupchat, so I set FreshRSS to send news items of a given feed or selected feeds to node `urn:xmpp:microblog:0` of groupchat `xmpp:syndication@conference.movim.eu?join`.

**Reply by Alkarex** [reply]
This might be a job for a FreshRSS extension.
As a first step, it would be good to find what library or code to build upon.
For instance https://github.com/Norgul/xmpp-php (does not seem maintained). Also a draft of XMPP to RSS gateway https://github.com/WST/rss-gateway

**Reply by sjehuda** [reply]
> This might be a job for a FreshRSS extension.

It definitely might!

> As a first step, it would be good to find what library or code to build upon.

Generally, I am not sure what library to select. There is a discussion on that regard at https://github.com/xenocrat/chyrp-lite/issues/250 (This is related to incorporating CMS with PubSub).

If xmpp-php supports [PubSub](https://github.com/Norgul/xmpp-php/issues/24), then it might do the job.

> Also a draft of XMPP to RSS gateway

I am not fond of RSS, which is not a standard and has [too many variation](https://web.archive.org/web/20060409104917if_/http://diveintomark.org:80/archives/2004/02/04/incompatible-rss). I think [Atom](https://github.com/edhelas/feedcleaner) is the best format and it is also the choice for Atom Over XMPP which is implemented on the Movim platform.

**Reply by Alkarex** [reply]
I meant RSS as the superset concept of RSS / ATOM

**Reply by sjehuda** [reply]
There is a PHP module which is called JAXL, and the repository includes examples as well.

* https://github.com/jaxl/JAXL
* https://jaxl.readthedocs.io/
* [http://abhinavsingh.com](https://web.archive.org/web/20160312204340if_/http://abhinavsingh.com) (jaxl.net

**Reply by Alkarex** [reply]
That library is not maintained anymore
https://github.com/jaxl/JAXL/issues/101#issuecomment-702558659

**Reply by sjehuda** [reply]
Yes. Indeed. Yet, it was the most used software library to create XMPP bots and other automated services, and it has support and examples for XMPP PubSub, so I deem it to be worth to try.

**Reply by Alkarex** [reply]
PR welcome for a FreshRSS extension if anyone is motivated in developing and maintaining it. I can help from the core if anything is lacking there.
See related work-in-progress https://github.com/FreshRSS/Extensions/pull/324

**Comment by sjehuda** [comment]
For record and reference.

There are already multiple parties that are interested in this feature.

The main concern, is a current and maintained PHP module for that task.

Tiki Wiki CMS Groupware

https://doc.tiki.org/xmpp

Chyrp Lite

https://github.com/xenocrat/chyrp-lite/issues/250
